source/glimmer-components/index.md

```diff
@@ -0,0 +1,37 @@
+## HTML elements as an abstraction for text documents
+
```

Comment:
While I understand that this is supposed ease the way in to the components, but throughout these guides there are only touches on "why components" or "why is this model good". I feel like it might be worth stating why components and how they fit into the rest of the system. Maybe it doesn't need to be in this document, but I think should follow instead of going right into refactoring.